pkg/client/client.go

```diff
@@ -439,6 +439,31 @@ func (c *KpmClient) CompileTarPkg(tarPath string, opts *opt.CompileOptions) (*kc
 	return c.CompileWithOpts(opts)
 }
 
+// CompileGitPkg will compile the kcl package from the git url.
```

Comment:
Please add test case for `CompileGitPkg` in https://github.com/kcl-lang/kpm/blob/main/pkg/client/client_test.go

You can use this repository as a test case in your tests,

- https://github.com/KusionStack/catalog

And you can select the following samples as the compiled entry

- https://github.com/KusionStack/catalog/tree/main/models/samples

You can refer to these to complete your test case：

- https://github.com/kcl-lang/kpm/blob/194168ae2db904ec56030379c3edcd4f7da803d8/pkg/client/client_test.go#L376
- https://github.com/kcl-lang/kpm/blob/194168ae2db904ec56030379c3edcd4f7da803d8/pkg/client/client_test.go#L821

Reply:
Hey @zong-zhe,
could you please give the final command to execute to run the github repo ? 
like `kpm run https://github.com/org/repo`

Currently cloning this for example https://github.com/KusionStack/catalog/tree/main/models/samples/hellocollaset/prod
would complain it is not a git repository.  

Thanks

Reply:
Hi @AkashKumar7902, 😃

Let me provide some more details about this comment.

We have currently decided to depart the `kpm cli`, and use a new command line [`kcl cli`](https://github.com/kcl-lang/cli).
All the command line in this repository will be deprecated. And the code in this repository will serve as a third-party library to provide basic functionality for Repository the `kcl cli`, while Repository `kcl cli` will be responsible for all command line parts.

So, the function `kpm run https://github.com/org/repo` does not need to be provided in this repository. In this repository, you only need to provide an API to compile KCL programs from Git Repo, which is in the PR you developed. You have done an excellent job on this ! 👍👍👍

The meaning of my comment is that you still need to write some unit test cases for your API.

There are currently a few issues to note:
* At present, the function provided by kpm to download GitHub repositories cannot download a certain directory in a repository, it must download the entire repository. So in your test cases, you need to use https://github.com/KusionStack/catalog as the repository Url, not its subdirectory https://github.com/KusionStack/catalog/tree/main/models/samples/hellocollaset/prod 
* At that time, there were no KCL files in the root directory of the repository, so it could not be compiled directly after downloading. It needs to be compiled in a subdirectory with KCL files. You can set the file you want to compile to `entry`. 
Therefore, you only need to test the API you wrote with a test case similar to the one below. Of course, the one given below is just pseudocode, you need to write more cases.

```
// The pseudo-code here
func test() {
 	CompileGitPkg(
	    CloneOption {
		Url: “https://github.com/KusionStack/catalog”
		tag: “0.1.0”
 	    }, 
	    opt.CompileOptions {
		entries: [“models/samples/hellocollaset/prod/main.k”]
	    }
    )
}
```

Reply:
Done 👍
I am sorry for being late was suffering from cold